test/network_analysis/network_properties.jl

```diff
@@ -649,286 +320,189 @@ let
-    rates1 = [:k1=>1.0, :k2=>1.0, :k3=>1.0, :k4=>1.0, :k5=>1.0, :k6=>1.0]
-    @test Catalyst.isdetailedbalanced(rn, rates1) == true
-    rates2 = [:k1=>2.0, :k2=>1.0, :k3=>1.0, :k4=>1.0, :k5=>1.0, :k6=>1.0]
-    @test Catalyst.isdetailedbalanced(rn, rates2) == false 
-end
+    @test Catalyst.fluxmat(MAPK, ratemap) == Catalyst.fluxmat(MAPK, ratevec) == Catalyst.fluxmat(MAPK, ratetup)
+    K = Catalyst.fluxmat(MAPK, ratemap; sparse = true)
+    @test Catalyst.issparse(K)
+    
+    A_k = Catalyst.laplacianmat(MAPK, ratemap)
+    @test all(col -> sum(col) == 0, eachcol(A_k))
+    A_k = Catalyst.laplacianmat(MAPK, ratemap; sparse = true)
+    @test Catalyst.issparse(A_k)
 
-# Independent cycle tests: the following reaction entwork has 3 out-of-forest reactions. 
-let
-    rn = @reaction_network begin
-        (k1, k2), A <--> B + C
-        (k3, k4), A <--> D
-        (k5, k6), B + C <--> D
-        (k7, k8), A + D <--> E
-        (k9, k10), G <--> 2F
-        (k11, k12), A + D <--> G
-        (k13, k14), G <--> E
-        (k15, k16), 2F <--> E
-        (k17, k18), A + E <--> H
+    numeqs = similar(eqs)
+    for i in 1:length(eqs)
+        numeqs[i] = substitute(eqs[i], ratemap)
     end
-
-    rcs, D = reactioncomplexes(rn)
-    k = rand(rng, numparams(rn))
-    p = parameters(rn)
-    rates = Dict(zip(parameters(rn), k))
-    @test Catalyst.isdetailedbalanced(rn, rates) == false
-
-    # Adjust rate constants to obey the independent cycle conditions. 
-    rates[p[6]] = rates[p[1]]*rates[p[4]]*rates[p[5]] / (rates[p[2]]*rates[p[3]]) 
-    rates[p[14]] = rates[p[13]]*rates[p[11]]*rates[p[8]] / (rates[p[12]]*rates[p[7]])
-    rates[p[16]] = rates[p[8]]*rates[p[15]]*rates[p[9]]*rates[p[11]] / (rates[p[7]]*rates[p[12]]*rates[p[10]])
-    @test Catalyst.isdetailedbalanced(rn, rates) == true
+    @test all(iszero, simplify(numeqs - S*K*Φ))
+    @test all(iszero, simplify(numeqs - Y*A_k*Φ))
 end
 
-# Deficiency two network: the following reaction network must satisfy both the independent cycle conditions and the spanning forest conditions 
+# Test handling for weird complexes and combinatoric rate laws. 
 let
     rn = @reaction_network begin
-        (k1, k2), 3A <--> A + 2B
-        (k3, k4), A + 2B <--> 3B
-        (k5, k6), 3B <--> 2A + B
-        (k7, k8), 2A + B <--> 3A
-        (k9, k10), 3A <--> 3B
-    end
-
-    rcs, D = reactioncomplexes(rn)
-    @test Catalyst.edgeindex(D, 1, 2) == 1
-    @test Catalyst.edgeindex(D, 4, 3) == 6
+        k1, 2X + Y + 3Z --> ∅
+        (k2, k3), 2Y + 2Z <--> 3X
+    end
+
+    Φ = Catalyst.massactionvector(rn); specs = species(rn)
+    crvec = [rn.X^2/2 * rn.Y * rn.Z^3/6,
+             1.,
+             rn.Y^2/2 * rn.Z^2/2,
+             rn.X^3/6]
+    @test isequal(Φ, crvec)
+    ncrvec = [rn.X^2 * rn.Y * rn.Z^3,
+              1.,
+              rn.Y^2 * rn.Z^2,
+              rn.X^3]
+    Φ_2 = Catalyst.massactionvector(rn; combinatoric_ratelaws = false)
+    @test isequal(Φ_2, ncrvec)
```

Comment:
Should these kind of tests be `issetequal`? We don't generally want to make assumptions about ordering.

Reply:
Thinking about this more I think we have to check the order for the output to the mass action vector, otherwise building the ODEs with something like (Y*K*\Phi) will not yield correct results. In general the set of ODEs has to be in the iteration order of `species(rn)`.